### PR TITLE
Fix tcpdump report error when tacacs enabled.

### PR DESCRIPTION
**Pull request by liuh-80**
Fix tcpdump report error when tacacs enabled.

#### Why I did it
Fix tcpdump report error when tacacs enabled:
Sep  1 09:25:18.189395 vlab-01 ERR tcpdump: nss_tacplus: /etc/tacplus_nss.conf fopen failed
Sep  1 09:25:18.189606 vlab-01 ERR tcpdump: nss_tacplus: bad config or server line for nss_tacplus

This is because debian add a patch create AppArmor profile for resource access control. The profile need update to allow tcpdump access /etc/tacplus_nss.conf.

##### Work item tracking
- Microsoft ADO: 17667308

#### How I did it
Modify tcpdump AppArmor profile, add new line to allow tcpdump access TACACS config file:

  /etc/tacplus_nss.conf r,

#### How to verify it
Pass all UT.
Manually verify error fixed.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [x] 202205
- [x] 202211
- [x] 202305

#### Tested branch (Please provide the tested image version)

- [x]  SONiC.master-16372.360223-d5d6c64c5
- [X] SONiC.202205-16610.367814-d52a71b48
- [X] SONiC.202211-16611.367823-11a8b8d55

#### Description for the changelog
Fix tcpdump report error when tacacs enabled.

### CODE REVIEW

**Review by qiluo-msft — APPROVED** [review]

**Comment by lguohan** [comment]
is this tested on 202205 branch?

**Comment by lguohan** [comment]
why tcpdump needs access to /etc/tacplus_nss.conf?

**Comment by lguohan** [comment]
the apparmor has local/usr.bin.tcpdump to allow local customization, suggest to use that mechanism instead of modifying this file which is a little bit fragile.

**Review by lguohan — CHANGES_REQUESTED** [review]
suggest to override /etc/apparmor.d/local/usr.bin.tcpdump

**Comment by liuh-80** [comment]
> is this tested on 202205 branch?

Yes, this tested on 202205 branch by copy the modified profile.

**Comment by liuh-80** [comment]
> tacplus_nss

The code of tcpdump will invoke 'getpwnam' API, and nss-tacplus plugin hook this API for TACACS authentication.

**Comment by liuh-80** [comment]
> suggest to override /etc/apparmor.d/local/usr.bin.tcpdump
Will update this PR later.

File: build_debian.sh [hunk 1]
```diff
@@ -395,6 +395,12 @@ sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y in
 	lsof                    \
 	sysstat
 
+# latest tcpdump control resource access with AppArmor.
+# override tcpdump profile to allow tcpdump access TACACS config file.
+if [ -f ${FILESYSTEM_ROOT}/etc/apparmor.d/local/usr.bin.tcpdump ]; then
+    sudo LANG=C chroot $FILESYSTEM_ROOT /bin/bash -c "echo -e '\n# for access tacacs config\n/etc/tacplus_nss.conf r,\n' > /etc/apparmor.d/local/usr.bin.tcpdump"
```
Comment (lguohan) [inline build_debian.sh:401]: there is local file why to add some files in the buildimage, this does not seem to be the standard way.

Comment (liuh-80) [reply build_debian.sh:401]: This code is modifying the local file: '/etc/apparmor.d/local/usr.bin.tcpdump'
The local file been created by tcpdump install.

File: build_debian.sh [hunk 2]
```diff
@@ -395,6 +395,12 @@ sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y in
 	lsof                    \
 	sysstat
 
+# latest tcpdump control resource access with AppArmor.
+# override tcpdump profile to allow tcpdump access TACACS config file.
+if [ -f ${FILESYSTEM_ROOT}/etc/apparmor.d/local/usr.bin.tcpdump ]; then
```
[the remark this answers is not in the document]
Comment (liuh-80) [reply build_debian.sh:400]: The tcpdump package will always create apparmor local profile with dh_apparmor command.
So, this is by design, if the local profile does not exist, we don' need create it.

**Review by qiluo-msft — DISMISSED** [review]

File: build_debian.sh (hunk 1 above)
Comment (liuh-80) [reply build_debian.sh:401]: @lguohan , Could you please check this PR again? I reply your comments, current code does not add new file in buildimage, when install tcpdump, the package will install a local apparmer profile, and current code will check and modify the local apparmer profile.

**Comment by qiluo-msft** [comment]
@liuh-80 As the process, if you require backport to 202205, please make sure it could be clean cherry-pick to 202205 locally, test it locally, and add the version string into PR description "Tested branch (Please provide the tested image version)". This PR description process only applies to the earliest branch you want to backport.

---
In reply to: [1712434300](https://github.com/sonic-net/sonic-buildimage/pull/16372#issuecomment-1712434300) [](http://example.com/codeflow?ancestors=1712434300)

File: build_debian.sh (hunk 1 above)
Comment (lguohan) [reply build_debian.sh:401]: the local apparmer profile is empty, right?

it is better to make the local file more deterministic 

suggest to use this approach.

sudo cp files/dhcp/rfc3442-classless-routes $FILESYSTEM_ROOT/etc/dhcp/dhclient-exit-hooks.d

Comment (liuh-80) [reply build_debian.sh]: @lguohan , thanks, fixed according to your suggestion, please check.

**Comment by liuh-80** [comment]
> @liuh-80 As the process, if you require backport to 202205, please make sure it could be clean cherry-pick to 202205 locally, test it locally, and add the version string into PR description "Tested branch (Please provide the tested image version)". This PR description process only applies to the earliest branch you want to backport.
> 
> In reply to: [1712434300](https://github.com/sonic-net/sonic-buildimage/pull/16372#issuecomment-1712434300)

Fixed, verify this PR can clean cherry-pick to 202205, also tested image version updated.

**Review by lguohan — APPROVED** [review]

**Comment by StormLiangMS** [comment]
@liuh-80 have you tested with 202305? Could you update in the description?

**Comment by liuh-80** [comment]
> @liuh-80 have you tested with 202305? Could you update in the description?

Verified with SONiC.202305.388563-584c448b2, PR description updated.

**Pull request merged**